velox/exec/tests/DeleteTest.cpp

```diff
@@ -0,0 +1,325 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/dwio/common/tests/utils/BatchMaker.h"
+#include "velox/exec/tests/utils/OperatorTestBase.h"
+#include "velox/exec/tests/utils/PlanBuilder.h"
+#include "velox/exec/tests/utils/QueryAssertions.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::test;
+using namespace facebook::velox::exec::test;
+
+namespace {
+
+const std::string kMockConnectorId = "mock";
+const vector_size_t kMaxBatchSize = 1024;
+
+class MockTableHandle : public connector::ConnectorTableHandle {
```

Comment:
Indeed, it should be. I noticed this is also what other operator tests did. 
Actually I have tried before this PR, but gave it up. 😭

I believe it’s a huge project to make Hive connector have DELETE support In VELOX. Maybe we should create another umbrella issue to track it and add the support gradually.
It’s corresponding to the file format(Parquet, ORC), table format(where to maintain the metadata, RowID column support, etc)， the way of doing the deletion(Copy-On-Write or Base + Delta, Merge-on-Read) and the reader logic needs refactor to adapter to the updatable data source too AFAIK.

As the result, I used this mock connector to build the tests. 
The test cases are designed as the integrated tests in the scope of the pipeline.

Reply:
@dingqiaoyi Thank you for explaining your thought process. I have a follow-up question. Do you have a design for adding delete support to Hive connector? If not, how do you plan to use the new functionality? It would be nice to have a design for the full feature hashed out before starting to implement individual pieces.

Reply:
@dingqiaoyi There are these open Data Lake Table formats like Delta, Hudi, Iceberg, etc. that support transactions and will need a different version of delete/update operators. Presto has integration with these open formats. I think it will be nice to consider the requirement there and come up with a design that is not specific to PrestoDB.

Reply:
I see that you already mention these formats above `deletion(Copy-On-Write or Base + Delta, Merge-on-Read)`.
You also mention `the reader logic needs refactor to adapter to the updatable data source too AFAIK.`  I am trying to understand this.
You explain `We need an abstraction of UpdatableDataSource, which keeps both the scanning interfaces (next ...) and also the delete/update interfaces (delete/updateRows).`. I am lost here. Can you give some more details?
Since Parquet files are immutable, new files are created with the deleted/updated data.

Reply:
@dingqiaoyi 

> I’m now work in progress to integrate VELOX with a private developing lakehouse-like storage service which includes the DML supporting work.

I'm curious what's the name of the service you are working on and whether it is integrated with Spark, Presto or some other query engine. Do you have the code available somewhere on GitHub?

Reply:
CC: @gggrace14 @spershin